crates/ty_python_semantic/resources/mdtest/narrow/type_guards.md

```diff
@@ -215,23 +214,19 @@ class C(Generic[T]):
     v: T
 
 def _(a: tuple[Foo, Bar] | tuple[Bar, Foo], c: C[Any]):
-    # TODO: Should be `TypeGuard[Foo @ a[1]]`
-    if reveal_type(guard_foo(a[1])):  # revealed: @Todo(`TypeGuard[]` special form)
+    if reveal_type(guard_foo(a[1])):  # revealed: TypeGuard[Foo @ a[1]]
         # TODO: Should be `tuple[Bar, Foo]`
```

Comment:
Orthogonal to this PR, but I'm not sure this should even be a TODO. No other type checker draws this conclusion, and I think the odd semantics of `TypeGuard` make it unclear whether it's even valid to do so.
```suggestion
```